### DISCUSSION TITLE
How to load ArcGIS mapserver layer in leafmap?

### DISCUSSION THREAD

**Discussion by lzw19910** [opengeos/leafmap#282] [Q&A] [answered]
I need to add `ArcGIS Tiled mapserver layer` and `ArcGIS mapserver layer`  into leafmap.
First, I success add `ArcGIS Tiled mapserver layer` in jupyter notebook using the sample service [world_imagery](https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer). And the tile map successfully shows in map.
```
import leafmap.foliumap as leafmap
m = leafmap.Map()
m.add_tile_layer('https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}','name','attribution')
m
```
![image](https://user-images.githubusercontent.com/42457098/188305834-e867a8d2-17f2-4cbc-a246-fdbd813d854d.png)

But when i add `ArcGIS mapserver layer` using the sample service [Military](https://sampleserver6.arcgisonline.com/arcgis/rest/services/Military/MapServer). Although i can see the layer name in layer managemet bottom, but the map just load the basemap. The service feature dose not show in the map.
```
import leafmap.foliumap as leafmap
m2 = leafmap.Map()
m2.add_tile_layer('https://sampleserver6.arcgisonline.com/arcgis/rest/services/Military/MapServer/{z}/{y}/{x}','military','attribution')
m2
```
![image](https://user-images.githubusercontent.com/42457098/188305765-d494b988-5a51-4aa9-a4de-e0bc3b25c971.png)

Maybe it's wrong to use `m2.add_tile_layer()`,or `......MapServer/{z}/{y}/{x}` is wrong. I look through the document (https://leafmap.org/) but still cannot find solution.
Please tell me how to right load ArcGIS mapserver layer in leafmap if you know. Thank you so much for your assistance, best wishes!!!

**Comment by giswqs** [comment, marked as answer]
The military layer does not support WMTS. It won't work with leaflet. 
![image](https://user-images.githubusercontent.com/5016453/188314648-95f2b45d-12ee-436e-88ce-3da8654d4892.png)


![image](https://user-images.githubusercontent.com/5016453/188314675-61b27b93-f138-4f80-ac4a-59164f5ae2ba.png)
